### DISCUSSION TITLE
Kinematic Character Controller?

### DISCUSSION THREAD

**Discussion by cuppajoeman** [jrouwe/JoltPhysics#853] [Q&A] [answered]
I was reading through the comment in the character class, it reads:

> Runtime character object. This object usually represents the player or a humanoid AI. It uses a single rigid body, usually with a capsule shape to simulate movement and collision for the character. The character is a keyframed object, the application controls it by setting the velocity.

Then I found out that the character is being created as a dynamic rigid body, which means that physics are being simulated for the body.

https://github.com/jrouwe/JoltPhysics/blob/a2b6361777f403bd219a89666b61fb92dad51cce/Jolt/Physics/Character/Character.cpp#L36

Also if issues like #717 apply to dynamic rigid bodies, wouldn't the character also suffer from random jumping when moving along surfaces?

If we take a look at what physX has to say about the topic we can see that there are various drawbacks to using a dynamic character controller: https://docs.nvidia.com/gameworks/content/gameworkslibrary/physx/guide/Manual/CharacterControllers.html. 

The bullet point there that worries me would be the inability to change the position of the character, which is very important for re spawning players.

Since Jolt seems to be using a dynamic character controller, I was wondering if there is anything done to combat these issues. 

Also to anyone who has actually used the character controller included with jolt have you noticed any major drawbacks or does it act like a kinematic character controller for the most part?

**Comment by jrouwe** [comment, marked as answer]
Take a look at CharacterVirtual...

**Reply by cuppajoeman** [reply]
Perfect, I assumed that was some interface class, but this is exactly what I was looking for.👍
